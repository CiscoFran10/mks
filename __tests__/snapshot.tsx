import React from 'react';
import Home from '@/app/page';
import { act } from 'react-dom/test-utils';
import { RenderResult, render } from '@testing-library/react';
import DefaultLayout from '../layouts/default-layout';

it('renders homepage with products', async () => {
  let container!: RenderResult;

  await act(async () => {
    container = render(
      <DefaultLayout>
        <Home />
      </DefaultLayout>
    );
  });

  expect(container).toMatchSnapshot();
});
