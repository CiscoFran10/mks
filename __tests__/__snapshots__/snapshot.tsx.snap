// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage with products 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bkbjWr dfCzRf"
      >
        <header
          class="sc-hBMVwJ lenMzg"
        >
          <a
            class="sc-fnVZQs ezPvZV"
            href="/"
          >
            MKS
            <span>
              Sistemas
            </span>
          </a>
          <button
            aria-haspopup="true"
            aria-label="Abrir carrinho"
            class="sc-fFSPgy jQNYqd"
          >
            <span />
            <span
              aria-label="Produtos no carrinho: 0"
              aria-live="polite"
            >
              0
            </span>
          </button>
        </header>
        <main
          class="sc-iemXqs ihKWDY"
        >
          <ul
            class="sc-bdnyFh jGaKwO"
          >
            <div
              class="sc-crzpnZ jdHURl"
            >
              <div
                class="image"
              />
              <div />
              <div />
            </div>
            <div
              class="sc-crzpnZ jdHURl"
            >
              <div
                class="image"
              />
              <div />
              <div />
            </div>
            <div
              class="sc-crzpnZ jdHURl"
            >
              <div
                class="image"
              />
              <div />
              <div />
            </div>
            <div
              class="sc-crzpnZ jdHURl"
            >
              <div
                class="image"
              />
              <div />
              <div />
            </div>
            <div
              class="sc-crzpnZ jdHURl"
            >
              <div
                class="image"
              />
              <div />
              <div />
            </div>
            <div
              class="sc-crzpnZ jdHURl"
            >
              <div
                class="image"
              />
              <div />
              <div />
            </div>
            <div
              class="sc-crzpnZ jdHURl"
            >
              <div
                class="image"
              />
              <div />
              <div />
            </div>
            <div
              class="sc-crzpnZ jdHURl"
            >
              <div
                class="image"
              />
              <div />
              <div />
            </div>
          </ul>
        </main>
        <footer
          class="sc-ksltVi ljcAKG"
        >
          <p>
            MKS sistemas © Todos os direitos reservados
          </p>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bkbjWr dfCzRf"
    >
      <header
        class="sc-hBMVwJ lenMzg"
      >
        <a
          class="sc-fnVZQs ezPvZV"
          href="/"
        >
          MKS
          <span>
            Sistemas
          </span>
        </a>
        <button
          aria-haspopup="true"
          aria-label="Abrir carrinho"
          class="sc-fFSPgy jQNYqd"
        >
          <span />
          <span
            aria-label="Produtos no carrinho: 0"
            aria-live="polite"
          >
            0
          </span>
        </button>
      </header>
      <main
        class="sc-iemXqs ihKWDY"
      >
        <ul
          class="sc-bdnyFh jGaKwO"
        >
          <div
            class="sc-crzpnZ jdHURl"
          >
            <div
              class="image"
            />
            <div />
            <div />
          </div>
          <div
            class="sc-crzpnZ jdHURl"
          >
            <div
              class="image"
            />
            <div />
            <div />
          </div>
          <div
            class="sc-crzpnZ jdHURl"
          >
            <div
              class="image"
            />
            <div />
            <div />
          </div>
          <div
            class="sc-crzpnZ jdHURl"
          >
            <div
              class="image"
            />
            <div />
            <div />
          </div>
          <div
            class="sc-crzpnZ jdHURl"
          >
            <div
              class="image"
            />
            <div />
            <div />
          </div>
          <div
            class="sc-crzpnZ jdHURl"
          >
            <div
              class="image"
            />
            <div />
            <div />
          </div>
          <div
            class="sc-crzpnZ jdHURl"
          >
            <div
              class="image"
            />
            <div />
            <div />
          </div>
          <div
            class="sc-crzpnZ jdHURl"
          >
            <div
              class="image"
            />
            <div />
            <div />
          </div>
        </ul>
      </main>
      <footer
        class="sc-ksltVi ljcAKG"
      >
        <p>
          MKS sistemas © Todos os direitos reservados
        </p>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
